File: game/src/main/java/com/scottlogic/hackathon/game/Collectable.java
package com.scottlogic.hackathon.game;

import lombok.AllArgsConstructor;
import lombok.Value;

/** A collectable item in the game. */
@Value
@AllArgsConstructor
public class Collectable {
  /**
   * @return The unique id of the current collectable item.
   */
  Id id;

  /**
   * @return the type of the current collectable item.
   */
  Type type;

  /**
   * @return The position of the collectable item.
   */
  Position position;

  /** Collectable item types */
  public enum Type {
    /** Item that generates a new player */
    PLAYER
  }
}

File: python-contestant/src/game/collectable.py
from dataclasses import dataclass, field

from game.position import Position


@dataclass(frozen=True)
class Collectable:
    """A collectable item in the game.

    Attributes:
        id (int): The unique id of the current collectable item.
        position (Position): The position of the current collectable item.
        type (str): The type of the current collectable item.
    """
    id: int
    position: Position = field(compare=False)
    type: str
